app/controllers/concerns/foreman_puppet/api/import_puppetclasses_common_controller.rb

```diff
@@ -74,7 +74,13 @@ def changed_environments
         end
 
         unless @environments.any?
-          render_message(_('No changes to your environments detected'))
+          # check if environment id/name was valid at all
+          if @env_id && (Environment.authorized(:view_environments).find_by(id: @env_id) || \
```

Comment:
this is wrong, what about when we do not pass the `env_id`?
Shouldn't we just detect whether the `@env_id` is set or not and tailor the message according to that?